File: src/main/java/com/reservapp/backend/model/Horario.java
package com.reservapp.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalTime;

@Entity
@Table(name = "horarios")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Horario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_empresa", nullable = false)
    private Empresa empresa;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    private Dia dia;

    @Column(nullable = false)
    private LocalTime apertura;

    @Column(nullable = false)
    private LocalTime cierre;

    public enum Dia {
        LUNES, MARTES, MIERCOLES, JUEVES, VIERNES, SABADO, DOMINGO
    }
}

File: reservapp_mobile/lib/models/horario.dart
class Horario {
  final int id;
  final int idEmpresa;
  final String dia;
  final String apertura;
  final String cierre;

  const Horario({
    required this.id,
    required this.idEmpresa,
    required this.dia,
    required this.apertura,
    required this.cierre,
  });

  factory Horario.fromJson(Map<String, dynamic> json) => Horario(
    id: json['id'] ?? 0,
    idEmpresa: json['idEmpresa'] ?? 0,
    dia: json['dia'] ?? '',
    apertura: json['apertura'] ?? '00:00:00',
    cierre: json['cierre'] ?? '00:00:00',
  );
}
